common/client/models.go

```diff
@@ -0,0 +1,17 @@
+package client
+
+type SendTxReturnCode int
+
+// Generalized client error types that dictates what should be the next action, depending on the RPC error response.
+const (
+	Successful              SendTxReturnCode = iota
```

Comment:
Are there cases where it is convenient for `Successful` to be the zero value? I'm worried it could lead to mistakes since e.g. an uninitialized return var looks like success.

Reply:
I'm supportive of using Successful to be a non-zero value.
And then calling code can panic if 0 is returned.

Reply:
I don't think introducing panics will add value here. Zero values are meant to be useful in Go - it is just brittle to have this one represent success.

Reply:
I could potentially swap Successful with Fatal, although it could potentially pose a risk as well since all states are sensitive. @jmank88  would it be better if I started enum from 1 instead of 0 or it's not preferred as a pattern in go?

Reply:
Yeah, if there is no clear case for a good zero value then you can just start from 1:
```suggestion
	Successful              SendTxReturnCode = iota + 1
```
Or maybe you'll want an unexported var to reference:
```suggestion
	none                       SendTxReturnCode = iota
	Successful              SendTxReturnCode
```